src/main/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWork.java

```diff
@@ -55,22 +60,39 @@ protected void doRun() {
 
     private void cleanCloud(ComputeEngineCloud cloud) {
         logger.log(Level.FINEST, "Cleaning cloud " + cloud.getCloudName());
-        List<Instance> remoteInstances = findRemoteInstances(cloud);
+        ComputeClientV2 clientV2 = ComputeClientV2.createFromComputeEngineCloud(cloud);
+        List<Instance> remoteInstances = findRunningRemoteInstances(clientV2);
         Set<String> localInstances = findLocalInstances(cloud);
+        if (!(localInstances.isEmpty() || remoteInstances.isEmpty())) {
+            updateLocalInstancesLabel(clientV2, localInstances, remoteInstances);
+        }
         remoteInstances.stream()
                 .filter(remote -> isOrphaned(remote, localInstances))
                 .forEach(remote -> terminateInstance(remote, cloud));
     }
 
     private boolean isOrphaned(Instance remote, Set<String> localInstances) {
-        String instanceName = remote.getName();
-        logger.log(Level.FINEST, "Checking instance " + instanceName);
-        return !localInstances.contains(instanceName);
+        if (localInstances.contains(remote.getName())) {
+            return false;
+        }
```

Comment:
(Logically should not actually be necessary given that we just called `updateLocalInstancesLabel`, but I guess that method does not mutate the `Instance` in the list to reflect the just-bumped timestamp.)

Reply:
right, maybe not good to refetch (one more network call) the instance again right?

Reply:
Right, I think we do need to do this check here given other code. Would be nice to mention why in a comment.